GTViewController/Classes/Nav Classes/GTFollowupViewController.m

```diff
@@ -100,4 +146,146 @@ - (void)transitionToThankYou {
+    UIViewAnimationCurve animationCurve;
+    NSTimeInterval animationDuration;
+    
+    [[userInfo objectForKey:UIKeyboardAnimationCurveUserInfoKey] getValue:&animationCurve];
+    [[userInfo objectForKey:UIKeyboardAnimationDurationUserInfoKey] getValue:&animationDuration];
+
+    [self animateViewWithVerticalDelta:viewVerticalDelta
+                     animationDuration:animationDuration
+                        animationCurve:animationCurve];
+}
+
+
+- (void) animateViewWithVerticalDelta:(CGFloat)verticalDelta animationDuration:(NSTimeInterval) animationDuration animationCurve:(UIViewAnimationCurve)animationCurve {
+    // Get animation info from userInfo
+    CGRect newViewFrame = self.view.frame;
+    
+    [UIView beginAnimations:nil context:nil];
+    [UIView setAnimationDuration:animationDuration];
+    [UIView setAnimationCurve:animationCurve];
+    
+    newViewFrame.origin.y -= verticalDelta;
+    newViewFrame.size.height += verticalDelta;
+    self.view.frame = newViewFrame;
+    
+    [UIView commitAnimations];
+}
+
+
+- (void) resetViewPosition:(NSNotification *) notification {
+    NSDictionary *userInfo = notification.userInfo;
+    
+    // Get animation info from userInfo
+    NSTimeInterval animationDuration;
+    UIViewAnimationCurve animationCurve;
+    CGRect myFrame = self.view.frame;
+    
+    [[userInfo objectForKey:UIKeyboardAnimationCurveUserInfoKey] getValue:&animationCurve];
+    [[userInfo objectForKey:UIKeyboardAnimationDurationUserInfoKey] getValue:&animationDuration];
+    
+    [UIView beginAnimations:nil context:nil];
+    [UIView setAnimationDuration:animationDuration];
+    [UIView setAnimationCurve:animationCurve];
+    
+    myFrame.origin.y = 0;
+    myFrame.size.height = self.originalHeight;
+    
+    self.view.frame = myFrame;
+    
+    [UIView commitAnimations];
+}
+
+- (CGFloat)calculateVerticalDeltaFromCurrentField:(UIView *)currentField toNextField:(UIView *)nextField {
+    return nextField.frame.origin.y - currentField.frame.origin.y;
+    
+}
+
+#pragma mark - UITextFieldDelegate methods
+
+- (BOOL)textFieldShouldReturn:(UITextField *)textField {
+    NSInteger nextTag = textField.tag + 1;
+    
+    UIResponder *nextResponder = [textField.superview.superview viewWithTag:nextTag];
+    
+    if (nextResponder) {
+        UITextField *nextActiveField = (UITextField*)nextResponder;
+        
+        CGFloat verticalDelta = [self calculateVerticalDeltaFromCurrentField:self.activeField.superview
+                                                                 toNextField:nextActiveField.superview];
+        
+        self.activeField = nextActiveField;
+        
+        [self animateViewWithVerticalDelta:verticalDelta animationDuration:0.3 animationCurve:UIViewAnimationCurveEaseOut];
+        
+        if (![nextActiveField.superview.superview viewWithTag:++nextTag]) {
+            [nextActiveField setReturnKeyType:UIReturnKeyDone];
```

Comment:
jk. I just read you next method. i'm not used to a different ordering of these methods